### DISCUSSION TITLE
weight parameter as edge weight or edge length

### DISCUSSION THREAD

**Discussion by peetal** [networkx/networkx#4538] [Q&A]
Hello all, 

I'm trying to use `nx.all_pairs_dijkstra` to compute weighted global efficiency. I'm confused by what the `weight` is when I build the graph -- should the weight be the "edge weight" (i.e., coupling strength) or "edge distance"? In other words, when constructing a graph using NetworkX, would a large weight indicate stronger connection between two nodes, or longer distance between two nodes? 

Thank you very much in advance.

**Comment by dschult** [comment]
Dijkstra is an algorithm for shortest paths. So whatever the weight is called, the resulting path will be the smallest sum of the weights. So a large weight denotes a father distance. That edge will be less likely to be chosen for the shortest path.

**Reply by peetal** [reply]
I see! Based on my current understanding, if I want to compute the weighted degree/clustering coefficient for each node in a graph, the weight should indicate the coupling strength between every two nodes. However, if I want to compute the shortest path, the weight should indicate the distance between every two nodes. Is my current understanding correct? 

Thank you very much!

**Comment by dschult** [comment]
Yes, each measure that involves edge weights has to decide whether a larger number means the nodes are closer or farther apart. And historically some measures choose one way and some choose the other. You can easily switch between the two by constructing the reciprocal of the edge weights. But it can be confusing sometimes. 

```python
nx.set_edge_attributes(G, {e: 1/d["distance"] for e, d in G.edges.items()}, "reciprocal_distance")
```
